virtcontainers/qemu.go

```diff
@@ -1275,6 +1275,59 @@ func (q *qemu) hotplugVhostUserDevice(vAttr *config.VhostUserDeviceAttrs, op ope
 	return nil
 }
 
+func (q *qemu) qomGetSlot(qomPath string) (uint8, error) {
+	addr, err := q.qmpMonitorCh.qmp.ExecQomGet(q.qmpMonitorCh.ctx, qomPath, "addr")
+	if err != nil {
+		return 0xff, err
+	}
+	addrf, ok := addr.(float64)
+	// XXX going via float makes no real sense, but that's how
+	// JSON works, and we should get away with it for the small
+	// values we have here
+	if !ok {
+		return 0xff, fmt.Errorf("addr QOM property of %s is %t not a number", qomPath, addr)
+	}
+	addri := uint8(addrf)
+
+	if (addri & 0x7) != 0 {
```

Comment:
What does this case correspond to? Why can't it be non-zero in our use case?

If we ever need it, it seems easy enough to make the function return `(uint8, uint8, err)` and add one extra `uint8` to its caller too. So I guess you probably had a good reason to consider it would always be zero.